packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -286,19 +295,47 @@ export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartP
     );
   }
 
+  // Adjusts the margin percentage, based on the width of the svg (inversly propotional)
+  private _adjustBarSpacing(): void {
+    const svgWidth = this.barChartSvgRef.current?.getBoundingClientRect().width;
+    if (svgWidth) {
```

Comment:
From what I understand we are trying to mix domain (x values) with range (svg width) which is causing this complexity.
Suppose the original svgWidth = 500px and there are 5 bars with values 10,20,30,40,100.
So originally each unit of x takes 500/(10+20+30+40+100)=2.5px

If each gap has to be of 2 px, that means we have 500 - 4*2 = 492 px left to represent the whole domain.
So the new equation becomes 492/(10+20+30+40+100) = 2.46px per unit of x + 2 px per gap.

This way we can definitevely ensure the bar spacing is always 2 px. Regardless of the bar stretching and compressing.

Lets discuss tomorrow if there are any issues in this approach.

Reply:
Can't think of any issues in this approach.
I ran the mathematics at my end also, the equations are more simple and easier to understand.
We can definitely use this mathematics imo. Thanks for the suggestion, will work on the neccessary modifications here